tests/integration/ingestion.bats

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env bats
+
+load ../_common/helpers
+load ../_common/c4gh_generate
+
+# CEGA_CONNECTION and CEGA_USERS_CREDS should be already set,
+# when this script runs
+
+function setup() {
+
+    # Defining the TMP dir
+    TESTFILES=${BATS_TEST_FILENAME}.d
+    mkdir -p "$TESTFILES"
+
+    # Test user
+    TESTUSER=dummy
+
+    # Find inbox port mapping. Usually 2222:9000
+    INBOX_PORT="2222"
+    # legarun docker port inbox 9000
+    # [ "$status" -eq 0 ]
+    # INBOX_PORT=${output##*:}
+    LEGA_SFTP="sftp -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -P ${INBOX_PORT}"
+}
+
+function teardown() {
+    rm -rf ${TESTFILES}
+}
+
+# Utility to ingest successfully a file
+function lega_ingest {
+    local TESTFILE=$1
+    local size=$2
+    local queue=$3
+    local inputsource=${4:-/dev/urandom}
+
+    # Generate a random file
+    legarun c4gh_generate $size $inputsource ${TESTFILES}/${TESTFILE}
+    [ "$status" -eq 0 ]
+
+    # Upload it
+    UPLOAD_CMD="put ${TESTFILES}/${TESTFILE}.c4ga /${TESTFILE}.c4ga"
+    legarun ${LEGA_SFTP} -i ${TESTDATA_DIR}/${TESTUSER}.sec ${TESTUSER}@localhost <<< ${UPLOAD_CMD}
+    [ "$status" -eq 0 ]
+
+    # Fetch the correlation id for that file (Hint: with user/filepath combination)
+    retry_until 0 100 1 ${MQ_GET} v1.files.inbox "${TESTUSER}" "/${TESTFILE}.c4ga"
+    [ "$status" -eq 0 ]
+    CORRELATION_ID=$output
+
+    # Publish the file to simulate a CentralEGA trigger
+    MESSAGE="{ \"user\": \"${TESTUSER}\", \"filepath\": \"/${TESTFILE}.c4ga\"}"
+    legarun ${MQ_PUBLISH} --correlation_id ${CORRELATION_ID} files "$MESSAGE"
+    [ "$status" -eq 0 ]
+
+    # Check that a message with the above correlation id arrived in the expected queue
```

Comment:
nope, not here.
If you want to make such a test, it's ok, but it'll have to be in another place.
I can think of putting it in the robustness part.
I'll explain more tomorrow.

Reply:
The idea behind such a db tests is that in unit tests we tested the db functions, but we have not checked that the data is registered as we expect in the actual PostgresDB, thus we need to check it either in ingestion (that is why I suggested here) or somewhere else

Reply:
In terms of the logic involved in testing, you don't check that the db receives the right value. You check that whichever component that receives/needs the value can do its function.
The value itself is irrelevant.

Assume that you change the component to request another value, and you also change the database to store the new value, then in that case, the functionality is there and all fine, but the test fails. The issue is in the test.

The test should tell you whether an _implementation_ matches a _specification_.
When the implementation changes, but not the specification, there is no reason to change the test. If this is needed, the test is flawed.

That said, there are some "checking" that can be done, to see if the database does store some appropriate selected values, and in that case, I'd say it's a unit test for the database, not an integration test. Integration tests are concerned with specifications, not implementations.

Reply:
that sounds more like Functional tests rather than Integration tests, I view integration tests as focused also on that components integrate with one another meaning looking that the internals are ok ... but then again these views are for a Slack discussion.